simulator/syncer/syncer.go

```diff
@@ -0,0 +1,280 @@
+	if filteringFn, ok := s.filteringFunctions[gvr]; ok {
+		if ok, err := filteringFn(ctx, resource, s.clients, Update); !ok || err != nil {
+			return err
+		}
+	}
+
+	// Run the mutating function for the resource.
+	if mutatingFn, ok := s.mutatingFunctions[gvr]; ok {
+		resource, err = mutatingFn(ctx, resource, s.clients, Update)
+		if err != nil {
+			return xerrors.Errorf("failed to mutate resource: %w", err)
+		}
+	}
+
+	// Create the resource on the destination cluster using the dynamic client
+	_, err = s.clients.DestDynamicClient.Resource(gvr).Namespace(namespace).Update(
+		ctx,
+		resource,
+		metav1.UpdateOptions{},
+	)
+	if err != nil {
+		return xerrors.Errorf("failed to create resource: %w", err)
+	}
+
+	return nil
+}
+
+// removeUnnecessaryMetadata removes the metadata from the resource.
+func removeUnnecessaryMetadata(resource *unstructured.Unstructured) *unstructured.Unstructured {
+	resource.SetUID("")
+	resource.SetGeneration(0)
+	resource.SetResourceVersion("")
+
+	return resource
+}
+
+func (s *Service) deleteResourceOnDestinationCluster(
+	ctx context.Context,
+	resource *unstructured.Unstructured,
+) error {
+	// Extract the GroupVersionResource from the Unstructured object
+	gvk := resource.GroupVersionKind()
+	gvr, err := s.findGVRForGVK(gvk)
+	if err != nil {
+		return err
+	}
+
+	// Namespaces resources should be created within the namespace defined in the Unstructured object
+	namespace := resource.GetNamespace()
+
+	// Create the resource on the destination cluster using the dynamic client
+	err = s.clients.DestDynamicClient.Resource(gvr).Namespace(namespace).Delete(
+		ctx,
+		resource.GetName(),
+		metav1.DeleteOptions{},
+	)
+	if err != nil {
+		return xerrors.Errorf("failed to delete resource: %w", err)
+	}
+
+	return nil
+}
+
+// findGVRForGVK uses the discovery client to get the GroupVersionResource for a given GroupVersionKind.
+func (s *Service) findGVRForGVK(gvk schema.GroupVersionKind) (schema.GroupVersionResource, error) {
+	m, err := s.clients.RestMapper.RESTMapping(gvk.GroupKind(), gvk.Version)
+	if err != nil {
+		return schema.GroupVersionResource{}, err
+	}
+
+	return m.Resource, nil
+}
+
+func (s *Service) addFunc(obj interface{}) {
+	ctx := context.Background()
+	unstructObj, ok := obj.(*unstructured.Unstructured)
+	if !ok {
+		klog.Error("Failed to convert runtime.Object to *unstructured.Unstructured")
+		return
+	}
+
+	err := s.createResourceOnDestinationCluster(ctx, unstructObj)
+	if err != nil {
+		klog.ErrorS(err, "Failed to create resource on destination cluster")
+	}
+}
+
+func (s *Service) updateFunc(_, newObj interface{}) {
+	ctx := context.Background()
+	unstructObj, ok := newObj.(*unstructured.Unstructured)
+	if !ok {
+		klog.Error("Failed to convert runtime.Object to *unstructured.Unstructured")
+		return
+	}
+
+	err := s.updateResourceOnDestinationCluster(ctx, unstructObj)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			klog.Info("Skipped to update resource on destination: ", err)
```

Comment:
```suggestion
			// We just ignore the not found error because the scheduler may preempt the Pods, or users may remove the resources for debugging.
			klog.Info("Skipped to update resource because we cannot find it in the destination cluster", "resource", klog.KObj(newObj))
```